sdk/ai/azure-ai-generative/azure/ai/generative/_telemetry/guide_to_logging.md

```diff
@@ -0,0 +1,106 @@
+## Configuring Client-side Logging for Azure Gen AI SDK
+
+
+This document explains the logging system for the Azure Generative AI SDK for developers in this repository who want to understand how the logging system works and how to configure it for their own methods and modules.
+
+
+
+### When do we log?
+
+Logging will be disabled if
+ - Calls are not being made from a Jupyter notebook. As an Azure SDK, azure-ai-generative must comply with the Azure SDK for Python's guidelines around logging. The guidelines limit logging to interactive scenarios only, meaning that client-side logging can only be enabled when a user is running SDK code from a Jupyter notebook.
+ - An incorrect user agent is passed. This prevents calls from other Azure SDKs being inadvertently logged under the azure-ai-generative namespace.
+ - The subscription ID is in the `test_subscriptions` list of internal developer subscription IDs declared in [`logging_handler.py`](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/ai/azure-ai-generative/azure/ai/generative/_telemetry/logging_handler.py). 
+ - An exception is triggered while trying to set up and return the logging handler. Logging should never interfere with the execution of the code, so if an exception is triggered, we simply move on.
+
+Logging can also be explicitly disabled in any scenario by setting the environment variable `os.environ["AZURE_AI_GENERATIVE_ENABLE_LOGGING"] = "True"` for azure-ai-generative or `os.environ["AZURE_AI_RESOURCES_ENABLE_LOGGING"] = "True"` for azure-ai-resources. (Note that since azure-ai-generative is dependent on azure-ai-resources, both environment variables will need to be set to false to prevent all logging when using azure-ai-generative.)
+
+In all of these cases, a `logging.NullHandler` is returned from `get_appinsights_log_handler`.
+
+### What do we log?
+
+The logging structure depends on access to the AIClient, which is passed to the operations class in the constructor. 
+
+If logging has been configured for a method, the following information will be logged:
+ - The method name
+ - The operation ID
```

Comment:
Does operation id apply to local running functions?

Reply:
Yes, for example, `QADataGenerator.export_to_file()` which just exports to the local file system has an operation ID in the appinsights logs when I run it.